### PR TITLE
Add dockerfile support and pre-commit +fixes

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
**In This PR**
- Hopefully updated the pattern matching to find and also the error checking
- Add in pre-commit
- changes to markdown spacing and spelling
- auto fixes to ts from pre-commit

**Pictures/videos**

- [ ] I've reviewed my own code
- [ ] Ive built and added the vsix

## To Do 

Parse the output from the running command to work inside of VScode. Fails when checkov finds an unsuppressed error in a Dockerfile

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,6 +68,7 @@
     "test": "node ./out/test/runTest.js"
   },
   "devDependencies": {
+    "@tootallnate/once": "^1.1.2",
```
Comment (elaygl) [inline package.json:71]: Are all the new packages needed?

Comment (JamesWoolfenden) [reply package.json:71]: it wouldnt build on my mac without them

Comment (JamesWoolfenden) [reply package.json:71]: i dont know if you have them added globally?

Comment (elaygl) [reply package.json:71]: For me it's working without them. @JamesWoolfenden can you please double verify?

Comment (JamesWoolfenden) [reply package.json]: removed.

**Comment by elaygl** [comment]
@JamesWoolfenden Can you please remove all the changes in the `package-lock.json` file?

**Review by elaygl — APPROVED** [review]

**Pull request merged**